### DISCUSSION TITLE
" bind: cannot assign requested address" while running core-command and device-virtual.

### DISCUSSION THREAD

**Discussion by divyashree123456** [edgexfoundry/community#91] [Q&A] [answered]
Hi All,
I am trying to start edgex-services using systemd and target files  to start all services  at auto-boot.while doing so i am facing error logs in Edegx-core-command and device-virtual services after booting the board as shown below :

**Apr 12 04:26:42 NB2-SOC-BSP-02FEB23-REL12-v1.2 core-command[207]: level=INFO ts=2023-04-12T04:26:42.009023Z app=core-command source=message.go:58 msg="Service started in: 66.142ms"
Apr 12 04:26:42 NB2-SOC-BSP-02FEB23-REL12-v1.2 core-command[207]: level=ERROR ts=2023-04-12T04:26:42.024215Z app=core-command source=httpserver.go:135 msg="Web server failed: listen tcp 10.168.1.185:59882: bind: cannot assign requested address"**

But when I manually start the services with "systemctl start Edgex-Core-Command.service" it was running fine .
please let me know for suggestions. 
Thanks

**Comment by iain-anderson** [comment, marked as answer]
https://www.freedesktop.org/wiki/Software/systemd/NetworkTarget/ may help
TL;DR:
```
After=network-online.target
Wants=network-online.target
```
in the .service file

**Comment by bnevis-i** [comment]
@divyashree123456

Please consider submitting a full set of systemd start scripts to edgex-examples.

**Reply by divyashree123456** [reply]
Hi @bnevis-i 
Shall I post  Edgex startup scripts for all Edgex-services as a contributor on edgex-examples repository ?
If yes ,please let me know ,how can i proceed with that.
Thanks

**Reply by bnevis-i** [reply]
Yeah, like in "deployment/systemd" or "deployment/shell" or something appropriate to describe your deployment.  And a good README.md explaining what it is and how it works.

**Reply by divyashree123456** [reply]
ok sure,Thanks

**Reply by divyashree123456** [reply]
Hi @bnevis-i 
I was unable to push my folder regrading systemd statup script for auto start of Edgex after boot to edgexfoundry/examples repository.
After commit ,i m facing " authentication failed " issue.so is there any prerequisites have to done before pushing.
please let me know for suggestions 

Thanks in advance.

**Reply by farshidtz** [reply]
You need to fork the repository and push to a branch on your own fork, then open a Pull Request against the main branch on the edgexfoundry/edgex-examples repo.

**Comment by farshidtz** [comment]
You may want to use the snaps which use systemd internally and define the right startup order. 

https://docs.edgexfoundry.org/3.0/getting-started/Ch-GettingStartedSnapUsers/

**Reply by divyashree123456** [reply]
Hi @farshidtz  
I have developed the startup script for edgex. Asked to push this as contributor,while doing so i m facing issue in pushing my code to main branch of edgexfoundry/examples repository.

**Reply by farshidtz** [reply]
This reply was in response to your original question about setting up the systemd services.

For the contribution issue, please see other answer on that thread https://github.com/orgs/edgexfoundry/discussions/91#discussioncomment-6174502

**Reply by bnevis-i** [reply]
@divyashree123456 Please see:

- https://docs.github.com/en/get-started/quickstart/github-flow
- https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide

You should also check your company policy regarding open source contributions, if there is any.
